Assets/Scripts/UI/InGame/Scoreboard/Scoreboard.cs

```diff
@@ -0,0 +1,195 @@
+
+        public static int SortByID (PhotonPlayer b, PhotonPlayer a)
+        {
+            return b.ID.CompareTo(a.ID);
+        }
+
+        public static int SortByName (PhotonPlayer b, PhotonPlayer a)
+        {
+            return RCextensions.returnStringFromObject(b.CustomProperties[PhotonPlayerProperty.name]).CompareTo(RCextensions.returnStringFromObject(a.CustomProperties[PhotonPlayerProperty.name]));
+        }
+
+        public static int SortByKills (PhotonPlayer a, PhotonPlayer b)
+        {
+            return RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.kills]).CompareTo(RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.kills]));
+        }
+
+        public static int SortByDeaths (PhotonPlayer a, PhotonPlayer b)
+        {
+            return RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.deaths]).CompareTo(RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.deaths]));
+        }
+        
+        public static int SortByMaxDamage (PhotonPlayer a, PhotonPlayer b)
+        {
+            return RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.max_dmg]).CompareTo(RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.max_dmg]));
+        }
+
+        public static int SortByTotalDamage (PhotonPlayer a, PhotonPlayer b)
+        {
+            return RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.total_dmg]).CompareTo(RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.total_dmg]));
+        }
+
+        public static int SortByScore (PhotonPlayer a, PhotonPlayer b)
+        {
+            var killsA = RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.kills]);
+            var deathsA = RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.deaths]);
+            var maxDamageA = RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.max_dmg]);
+            var totalDamageA = RCextensions.returnIntFromObject(a.CustomProperties[PhotonPlayerProperty.total_dmg]);
+
+            var killsB = RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.kills]);
+            var deathsB = RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.deaths]);
+            var maxDamageB = RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.max_dmg]);
+            var totalDamageB = RCextensions.returnIntFromObject(b.CustomProperties[PhotonPlayerProperty.total_dmg]);
+
+            float aScore = CalculateScore(killsA, deathsA, maxDamageA, totalDamageA);
+            float bScore = CalculateScore(killsB, deathsB, maxDamageB, totalDamageB);
+
+            return bScore.CompareTo(aScore);
+        }
+
+    #endregion
+
+        // Formula for calculating score using svork balance
+        private static float CalculateScore(int k, int d, int m, int t)
```

Comment:
Score should be calculated by a gamemode, as score depends on the gamemode. I.e. when someone plays capture gamemode, and they get 3 2k's, but haven't captured any objects, their score would be lower than someone who captured two objectives. For racing, the score would be the time, ect